build.gradle

```diff
@@ -455,4 +392,70 @@ installDist.dependsOn downloadGsaLibFile, downloadIntelDeflater
 build.dependsOn installDist
 check.dependsOn installDist
 
+model {
+    components {
+        VectorLoglessPairHMM(NativeLibrarySpec) {
+            // set location of source files
+            sources.cpp.source {
+                srcDirs "src/main/cpp/VectorLoglessPairHMM"
+                include "*.cc"
+            }
+
+            binaries.withType(SharedLibraryBinarySpec) { binary ->
+                if (System.env.GATK_SKIP_NATIVE_BUILD.toString().toBoolean()) {
+                    buildable = false
+                    println("INFO: \$GATK_SKIP_NATIVE_BUILD is true, skipping native AVX PairHMM library build.")
+                } else if (!org.gradle.internal.os.OperatingSystem.current().isLinux() && !org.gradle.internal.os.OperatingSystem.current().isMacOsX()) {
+                    buildable = false
+                    println("WARNING: Building the native AVX PairHMM library is not supported on this OS.")
+                } else {
+                    cppCompiler.args "-I", "${System.properties['java.home']}/../include"
+                    cppCompiler.args "-mavx"
+                    cppCompiler.args "-Wall"
+
+                    // build with gcov options on Travis
+                    if (System.env.CI.toString().toBoolean()) {
+                        cppCompiler.args "-g"
+                        cppCompiler.args "-O0"
+                        cppCompiler.args "--coverage"
+                        linker.args "--coverage"
+                    } else {
+                        cppCompiler.args "-O3"
+                    }
+
+                    if (org.gradle.internal.os.OperatingSystem.current().isMacOsX()) {
+                        cppCompiler.args "-I", "${System.properties['java.home']}/../include/Darwin"
+                    } else {
+                        cppCompiler.args "-I", "${System.properties['java.home']}/../include/linux"
+                        linker.args "-static-libgcc"
+                    }
+
+                    task copySharedLib(type: Copy) {
+                        from binary.sharedLibraryFile
+                        into "$compileJava.destinationDir/lib"
+                        dependsOn binary.tasks
```

Comment:
The only changes here should be that I moved the `copySharedLib.dependsOn binary.task`s into this block, and that we now have `dependsOn` statements for `shadowJar` and `sparkJar` as well as `jar`.  (I couldn't get it to for for `tasks.ofType(Jar)`) if someone knows more I'd be happy to have help.)